cla_public/apps/base/__init__.py

```diff
@@ -1,5 +1,17 @@
 "Base app"
-from flask import Blueprint
-
+import datetime
+from flask import Blueprint, after_this_request, request, redirect
 
 base = Blueprint('base', __name__)
+
+@base.before_app_request
+def detect_user_locale():
+    locale = request.args.get('locale')
+    
+    if locale is not None:
+        @after_this_request
+        def remember_locale(response):
+            expires = datetime.datetime.now() + datetime.timedelta(days=30)
+            response = redirect(request.path)
+            response.set_cookie('locale', locale, expires=expires)
```

Comment:
💭 We set the cookie without validating the input argument. As a follow-up, should we restrict the allowed inputs to the two known supported locales?

Reply:
I'm not sure if it's worth it. The current behaviour is that if the locale isn't recognised by babel, it defaults to 'en'.
Do you think there's a risk if we don't validate the input? If anything breaks, it would be because someone is playing with it.

Reply:
Hmm, I think the "path" selector sanitises and that's why frontend relies on a binary check:
https://github.com/ministryofjustice/cla_public/blob/0eef40a8ce87df94451ac5f95f4dcb6d210d6b62/cla_public/apps/base/views.py#L215-L216

I think it's a good idea to sanitize all user inputs. Cookies have a maximum size and people can be very clever to cause grief :)

Reply:
Will add this to a tech debt ticket as discussed.

Reply:
LGA-192